node/pkg/reporter/reporter.go

```diff
@@ -40,211 +36,94 @@ func NewReporter(ctx context.Context, opts ...ReporterOption) (*Reporter, error)
 
 	groupInterval := time.Duration(config.Interval) * time.Millisecond
 
-	topic, err := config.Ps.Join(topicString)
-	if err != nil {
-		log.Error().Str("Player", "Reporter").Err(err).Msg("Failed to join topic")
-		return nil, err
-	}
-
-	raft := raft.NewRaftNode(config.Host, config.Ps, topic, MESSAGE_BUFFER, groupInterval)
 	deviationThreshold := GetDeviationThreshold(groupInterval)
 	reporter := &Reporter{
```

Comment:
It doesn't have to be part of this pr, but can you make an update with the KaiaHelper declaration? 
I think same KaiaHelper object should be used for multiple reporter objects. 
current logic, which is using `SetKaiaHelper` function declares its own chainHelper, each reporter is going to have its own chainHelper which lead to using same wallets multiple times.

Reply:
and I'm guessing access to the `chainHelper` object should implement `mutex` right? otherwise we might end up with `nonce already used` error if multiple reporters try to make concurrent transaction submissions at the same time

Reply:
in edge cases it could be required, but i don't think it'll likely to happen since the submission is not done that much concurrently.

Reply:
if you see the logic inside chainHelper, it'll use nextReporter for subsequent transactions.

Reply:
if you want to keep it safe it'd be nice to have mutex though

Reply:
gotcha! i'll observe if that will happen

Reply:
let's just give some `time.Sleep` inside `startReporters()` iteration just in case, such as 1 second?

Reply:
implemented basic mutex

Reply:
🤔 not sure if this is the good approach, I was thinking that you were going to implement mutex into chainhelper package. normally that transaction function takes 3~5 seconds, meaning mutex lock will be done for quite a long time. worried if it'll be a drawback. 

it'd be nice if the part where retrieving nonce is locked

https://github.com/Bisonai/orakl/blob/50d01e323f9089c08930b282fec28c9c833c1f1c/node/pkg/chain/utils/utils.go#L252-L256

https://github.com/Bisonai/orakl/blob/50d01e323f9089c08930b282fec28c9c833c1f1c/node/pkg/chain/utils/utils.go#L333-L337

Reply:
if you don't want to handle chainHelper package, let's just remove mutex and give time.Sleep for delay between transaction request

Reply:
made pr related to this

https://github.com/Bisonai/orakl/pull/1875